src/util/ResourceUtil.ts

```diff
@@ -33,3 +38,14 @@ export function generateContainmentQuads(containerURI: NamedNode, childURIs: str
   return new RepresentationMetadata(containerURI,
     { [LDP.contains]: childURIs.map(DataFactory.namedNode) }).quads();
 }
+
+export async function cloneRepresentation(representation: Representation): Promise<BasicRepresentation> {
+  const data = await arrayifyStream(representation.data);
+  const result = new BasicRepresentation(
+    data,
+    new RepresentationMetadata(representation.metadata, representation.metadata.contentType),
+    representation.metadata.contentType,
```

Comment:
Setting the content-type is not necessary since it is part of the metadata. What is necessary is setting the `binary` flag correctly, which can be done by passing a boolean here instead of a string. So this can be replaced with `representation.binary`.